tox.ini

```diff
@@ -1,44 +1,39 @@
 [tox]
+isolated_build = True
 envlist =
     lint
     py37
     py38
     py39
     py310
 minversion = 3.14.2
-requires =
-    virtualenv >= 20.4.5
-    pip >= 21.1.1
 
 [testenv]
 passenv =
     LC_ALL
     LANG
     HOME
 commands =
-    pip install -e .
-    pytest --cov=cookiecutter --cov-report=term --cov-fail-under=100 --cov-branch {posargs:tests}
+    poetry install -vv --no-root
+    poetry run pytest --cov=cookiecutter --cov-report=term --cov-fail-under=100 --cov-branch {posargs:tests}
     cov-report: coverage html
     cov-report: coverage xml
-deps = -rtest_requirements.txt
-skip_install = true
+whitelist_externals = poetry
 
 [testenv:lint]
 commands =
-    python -m pre_commit run {posargs:--all}
-deps = pre-commit>=1.20.0
+    poetry install -vv --no-root
+    poetry run python -m pre_commit run {posargs:--all}
 usedevelop = false
 
 [testenv:docs]
 commands =
-    pip install -e .
-    pip install -r test_requirements.txt
-    pip install -r docs/requirements.txt
+    poetry install -vv --no-root
+    poetry run pip install -r docs/requirements.txt
     make docs
 whitelist_externals = make
 
 [testenv:safety]
 commands =
-    safety check --full-report
-deps =
-    safety
+    poetry install -vv --no-root
+    poetry run safety check --full-report
```

Comment:
Blank line at the end of file missed

Reply:
blank line added to the end of the file...